sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -511,7 +511,7 @@ class ContentDecodePolicy(SansIOHTTPPolicy):
     :type response_encoding: str
     """
     # Accept "text" because we're open minded people...
-    JSON_REGEXP = re.compile(r'^(application|text)/([0-9a-z+.]+\+)?json$')
+    JSON_REGEXP = re.compile(r'^(application|text)/([0-9a-z+.-]+\+)?json$')
```

Comment:
Do we need to constrain the "middle" part? What about:

^(application|text)/(.+)?\+json$

(I didn't check where we are using the captures, but if we are, they also look odd since we are capturing the '+')

Reply:
i guess that's fair. I know @lmazuel originally wrote the json regex, so not sure if he had a specific reason for making it more specific. I think for now, I can make the regex I use in autorest not constrain the middle part, and we can wait until @lmazuel is back to make it more generic on the core side. Thoughts?

Reply:
```
   Most HTTP header field values are defined using common syntax
   components (token, quoted-string, and comment) separated by
   whitespace or specific delimiting characters.  Delimiters are chosen
   from the set of US-ASCII visual characters not allowed in a token
   (DQUOTE and "(),/:;<=>?@[\]{}").

     token          = 1*tchar

     tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
                    / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
                    / DIGIT / ALPHA
                    ; any VCHAR, except delimiters
```

We should be careful about accepting anything. And @johanste , if you have the + outside of the parenthesis, you can't match `application/json` (the shorter you can catch is `application/+json`). Grouping don't matter, it's here just for the `?` at the end to say the subtype is optional. I would play safe and just add the `-`, though I won't argue if we think this is a maintenance risk.

Reply:
Yeah, missed the base case :)

So it either is "application/json", "text/json" (because we are nice) or it ends with (uses the suffix) "+json". Personally, I would probably not use a regex for this because regexes tend to make my python feel like perl.